### ISSUE TITLE
--all with babel-plugin-istanbul does not report coverage for files not dependents of the test entries

### ISSUE THREAD

**Issue by hershmire** [istanbuljs/nyc#911]
## Expected Behavior

When running NYC with the following [script command](https://github.com/hershmire/nyc-all/blob/master/package.json#L8):

```sh
NODE_ENV=test nyc mocha src/**/test-node/**/*.js
```

And with the following NYC configuration settings from package.json:

```json
"nyc": {
    "sourceMap": false,
    "instrument": false,
    "cache": false,
    "all": true,
    "exclude": [
      "**/test-node/**"
    ],
    "include": [
      "src/**"
    ],
    "reporter": [
      "json",
      "text",
      "text-summary"
    ],
    "require": [
      "babel-core/register"
    ],
    "extension": [
      ".jsx",
      ".js",
      ".mjs"
    ]
  },
```

And with the following .babelrc configuration setup:

```json
{
  "presets": [
    "babel-preset-react",
    ["babel-preset-env", {
      "targets": {
        "browsers": [
          "last 2 versions"
        ]
      }
    }]
  ],
  "env": {
    "test": {
      "plugins": [
        [
          "babel-plugin-istanbul",
          {
            "exclude": [
              "**/test-node/**"
            ]
          }
        ]
      ]
    }
  },
  "plugins": [
    "babel-plugin-transform-class-properties",
    "babel-plugin-transform-object-rest-spread"
  ]
}
```

I expect **all my files** that match the file extensions of .jsx, .js, and .mjs within the `src/**` directory will be included in the test coverage reports.

Below is what I would expect to see which I only get when I turn off babel-plugin-istanbul and let NYC instrument the files:

![screen shot 2018-08-23 at 15 50 04](https://user-images.githubusercontent.com/757524/44556586-48963c00-a6ef-11e8-841a-a32957239a35.png)

## Observed Behavior

The coverage reports only show the files that are dependencies to the tests. Here is the output I get:

![screen shot 2018-08-23 at 15 49 02](https://user-images.githubusercontent.com/757524/44556554-28ff1380-a6ef-11e8-8aa8-56cdd85b5eb3.png)

## Bonus Points! Code (or Repository) that Reproduces Issue

Here is a repository I created that illustrates this problem – https://github.com/hershmire/nyc-all

## Forensic Information

**Operating System:** the operating system you observed the issue on.
macOS High Sierra v10.13.6

**Environment Information:** information about your project's environment, see instructions below:

1. run the following script:

sh -c 'node --version; npm --version; npm ls' > output.txt

2. share a gist with the contents of output.txt.

https://github.com/hershmire/nyc-all/blob/master/output.txt

**Comment by coreyfarrell** [comment]
I've seen an issue like this, I believe the problem is that `"instrument": false` is preventing `"all": true` from being effective.  In some of my own repositories which use babel-plugin-istanbul I was able to get everything working by using:
```js
{
  "instrument": true,
  "sourceMaps": false,
  "all": true
}
```
Obviously that conflicts with what is documented.  I have not had a chance to investigate this further, the `"all": true` functionality definitely needs work.

**Comment by JaKXz** [comment]
@hershmire are you still experiencing this with the latest version of nyc and Corey's suggestion?

**Comment by CynicalBusiness** [comment]
I know this is a bit of an older issue, but I am still seeing this same behavior in another repository. This time, both on `Windows 10 Enterprise 1809` and `Ubuntu Desktop 18.04`.

I'm using the command `cross-env NODE_ENV=test DEBUG= nyc mocha` through `yarn`.

I've got the following:
- `@babel/core@^7.2.2`
- `babel-plugin-istanbul@^5.1.1`
- `nyc@^13.3.0`
- `istanbul@^0.4.5`
- `mocha@^6.0.2`

My `.nycrc.json`:
```json
{
  "include": [
    "src"
  ],
  "reporter": [
    "lcov",
    "text-summary"
  ],
  "require": [
    "@babel/register"
  ],
  "extension": [
    ".ts",
    ".tsx"
  ],
  "all": true,
  "instrument": false,
  "sourceMap": false
}
```

...and `babelrc.js`:
```javascript

const presets = [
  [ '@babel/env', {
    targets: {
      node: true
    }
  }],
  '@babel/typescript'
]
const plugins = [
  '@babel/proposal-class-properties',
  '@babel/proposal-object-rest-spread',
  [ 'transform-es2015-modules-commonjs', {
    noInterop: true
  }]
]

exports = module.exports = {
  comments: false,
  compact: true,
  env: {
    test: {
      plugins: [ 'istanbul' ]
    }
  },
  plugins,
  presets,
  sourceMaps: true
}
```

My directory structure, in short, looks like this:
```

src/
  app/
    index.ts
    (other TS sources)
    package.json
    tsconfig.json
  base/
    (same as above)
  core/
    (same)
  lib/
    (same)
test/
  (tests and such)
  mocha.opts
.babelrc.js
.nycrc.json
package.json
tsconfig.json
(plus some other files)
```

In short, I'm trying to cover every TS file in `src/`, even though (right now, anyway), my tests only require files in `lib/`. That directory gets covered fine, but the others are not; I get the exact same behavior reguardless of the setting of `all`.

Switching `instrument` to `true` made it now cover the `src/core` directory (which is the only one right now with no sub-directores), but it's still missing the other two. Additionally, including `src` versus `src/**` versus `src/**/*.ts` has no effect either.

This looks like the same behavior as the original, with the same configuration (that is, `@babel/register` and `babel-plugin-istanbul`)

**Comment by coreyfarrell** [comment]
@CynicalBusiness can you provide a link to a repository?

**Comment by 0x6431346e** [comment]
Really weird, but I managed to get it "working" by removing `require @babel/register` and setting `instrument` to true.

```diff
  "nyc": {
-    "require": [
-      "@babel/register"
-    ],
    "sourceMap": false,
-    "instrument": false,
+    "instrument": true,
    "all": true
  }
```

Would be awesome if someone could update the tutorial.

Note: I'm quoting working because I'm not sure if the lines of code are calculated correctly since I removed the require hook... :man_shrugging:
